Fix ttnn docs

Aswinmcw opened the pull request:
### Ticket
NA

### Problem description
ttnn docs are not good looking

### What's changed
Removed the fast operation preview and added actual parameters taken from the c++ pybind functions

Live preview : http://yyzeon16.local.tenstorrent.com:12345/ttnn/ttnn/api.html

## Eltwise
<p align="center">
  <img src="https://github.com/user-attachments/assets/b9d7a3b8-0d6f-403c-af3d-7b19928d5b3b" alt="Before Eltwise" width="48%" />
  <img src="https://github.com/user-attachments/assets/4b7e2349-98f4-4d4b-85fb-879534451b78" alt="After Eltwise" width="48%" />
</p>

## Matmul
<p align="center">
  <img src="https://github.com/user-attachments/assets/607939a5-3f6a-4891-8afc-0efd0c90b289" alt="Before Matmul" width="48%" />
  <img src="https://github.com/user-attachments/assets/37add8e6-220b-4bd6-92f9-216da6c35f2e" alt="After Matmul" width="48%" />
</p>

## CCL
<p align="center">
  <img src="https://github.com/user-attachments/assets/8eb420ea-d4be-4e39-ba92-4e4b9b2acff2" alt="Before CCL" width="48%" />
  <img src="https://github.com/user-attachments/assets/f336de94-628d-447b-857e-32663fad2b31" alt="After CCL" width="48%" />
</p>

## conv2d
<p align="center">
  <img src="https://github.com/user-attachments/assets/3dd71811-01be-4905-98ec-df92ee836198" alt="Before conv2d" width="48%" />
  <img src="https://github.com/user-attachments/assets/923ee4e9-307b-4a30-b8f3-dc8b953ffbbe" alt="After conv2d" width="48%" />
</p>


## max_pool2d
<p align="center">
  <img src="https://github.com/user-attachments/assets/3ae72a30-2a94-4f6c-a2cd-54f9dcb7598c" alt="Before max_pool2d" width="48%" />
  <img src="https://github.com/user-attachments/assets/773712e9-8173-41d8-a23e-728e8490640d" alt="After max_pool2d" width="48%" />
</p>


### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

Aswinmcw committed:
Add custom documenters for Operation and FastOperation in Sphinx

- Introduced `OperationDocumenter` and `OperationDataDocumenter` to enhance documentation capabilities for `Operation` and `FastOperation` decorators.
- Updated `setup` function to register new documenters.
- Improved parameter representation in `Param` class to conditionally display optional types.
- Added `__repr__` methods to `FastOperation` and `Operation` classes for better debugging output.

Aswinmcw committed:
Remove unused imports

aczajkowskiTT commented:
Looks great.

Aswinmcw committed:
Merge branch 'main' into Aswinmcw/fix_docs

Aswinmcw commented on ttnn/ttnn/decorators.py:
```diff
@@ -420,6 +423,9 @@ def __gt__(self, other):
         return hash(self.python_fully_qualified_name)
 
+    def __repr__(self):
+        return ""
```
Default behaviour is the fastoperation output which we are seeing currently

ntarafdar reviewed (approved):
this is beautiful thank you

bbradelTT commented:
@Aswinmcw for matmul/fused/reduce this duplicates info. Would you be able to remove what's in the pybind doc strings already, or would that need to be a separate PR?

Aswinmcw committed:
Merge branch 'main' into Aswinmcw/fix_docs

Aswinmcw committed:
Update documentation for various tensor operations in ttnn

- Enhanced docstrings for `all_gather`, `all_reduce`, `all_to_all_combine`, `all_to_all_dispatch`, `mesh_partition`, `reduce_scatter`, and `conv2d` functions to improve clarity and detail.
- Added examples and clarified argument descriptions for better usability.
- Removed redundant parameter descriptions to streamline documentation.

These changes aim to provide clearer guidance for users and improve the overall documentation quality across the operations.

Aswinmcw committed:
Refactor documentation strings for tensor operations in ttnn

- Updated docstrings for various tensor operations including `bcast`, `chunk`, `expand`, `permute`, `reshape`, `interleaved_to_sharded`, `reshard`, `sharded_to_interleaved`, and others to improve clarity and formatting.
- Removed redundant parameter descriptions and ensured consistent formatting across all operations.
- These changes enhance the usability and readability of the documentation for users.

Aswinmcw committed:
Enhance documentation for convolution operations in ttnn

- Updated docstrings for `conv1d` and `conv_transpose2d` to improve clarity and consistency in parameter descriptions and return values.
- Reformatted documentation for better readability, including the addition of `Args`, `Keyword Args`, and `Returns` sections.
- These changes aim to enhance user understanding and usability of the convolution operations.

Aswinmcw committed:
Merge branch 'main' into Aswinmcw/fix_docs

Aswinmcw committed:
Refactor documentation for all_to_all_dispatch and example operations

### What's changed
- Updated the docstring for the `all_to_all_dispatch` function to improve clarity and organization.
- Enhanced the documentation for example operations to include detailed argument and return descriptions.

### Checklist
- [x] Documentation improvements made for better understanding of operations.

Aswinmcw committed:
Merge branch 'main' into Aswinmcw/fix_docs

Aswinmcw committed:
Enhance documentation for tensor operations

### What's changed
- Improved docstrings for `clone`, `scatter`, and various unary operations to provide clearer descriptions of parameters, return values, and examples.
- Added mathematical representations in the documentation for unary backward operations to enhance clarity.

### Checklist
- [x] Documentation updates for better understanding of tensor operations.

Aswinmcw committed:
Merge branch 'main' into Aswinmcw/fix_docs

Aswinmcw committed:
Merge branch 'main' into Aswinmcw/fix_docs

Aswinmcw committed:
Update documentation
